task/move_test.go

```diff
@@ -95,3 +99,94 @@ func TestMoveCtx(t *testing.T) {
 
 	}
 }
+
+// MaybeMove tests implementation
+func (s *stub) mayBeFetch() (Thing, bool, error) {
+	if s.fetchErr == true {
+		return nil, false, errors.New("something went wrong fetching things")
+	}
+
+	if s.curr >= len(s.toFetch) {
+		return nil, false, nil
+	}
+
+	t := s.toFetch[s.curr]
+	s.curr++
+	return t, true, nil
+}
+
+func (s *stub) mayBePut(t Thing) error {
+	if s.putErr {
+		return errors.New("could not continue putting thing")
+	}
+
+	s.gotPut = append(s.gotPut, t)
+	return nil
+}
+
+func TestMaybeMove(t *testing.T) {
+	t.Run("no errors during fetch and put", func(t *testing.T) {
+		testCase := struct {
+			stub *stub
+			want []Thing
+		}{
+			stub: &stub{
+				toFetch: []Thing{1, 2, 3},
+			},
+			want: []Thing{1, 2, 3},
+		}
+
+		if err := MaybeMove(testCase.stub.mayBeFetch, testCase.stub.mayBePut); err != nil {
+			t.Errorf("MaybeMove() got err %v; want %v", err, nil)
+		}
+
+		if diff := cmp.Diff(testCase.want, testCase.stub.gotPut); diff != "" {
+			t.Errorf("MaybeMove() mismatch (-want +got):\n%s", diff)
+		}
+	})
+
+	testCases := []struct{
+		name string
+		stub *stub
+		want []Thing
+	}{
+		{
+			name: "errors in put",
+			stub: &stub{
+				// I discovered a bug when []Thing{1, 2} is supplied, the second
+				// execution of fetch runs, why? It seems the second call to fetch was
```

Comment:
Sometimes it will run and sometimes it won't, depending on the timing of the `close(quit)` (this is the nuance of concurrency and why it's difficult, because things are never exactly the same on each run).

I'm not sure why it panics. Can you give me the stack trace with all the goroutine information?